### ISSUE TITLE
False positives with text categorizer

### ISSUE THREAD

**Issue by H20Watermelon** [explosion/spaCy#4304]
## Your Environment

* Operating System: Win10
* Python Version Used: 3.7
* spaCy Version Used: 2.1.8
* Environment Information:

I trained a couple of textcat models on approximately 8000 passages (most of which are one or two sentences long (max token count is about 60) for binary classification ("PRESENT": a passage contains descriptions of stock information; "NOT_PRESENT": a passage does not contain stock information). One model used the `simple_cnn` mode and the other the default `ensemble` mode. 

When applying the models to the test set, I noticed that the models would predict "PRESENT" with very high probabilities (> 0.999) for passages that are not even remotely related to stock information. In fact, the models would give high probabilities to sentences with either garbled text or very short text. A couple of the examples are below:

- 3wqedsad \n sadsads
- (etc

Both of the passages above received 1.0 for the PRESENT class.

It also seems that punctuation/symbols have a huge influence on predictions. For example, in the second example above, if the left parenthesis is removed, the probability for PRESENT drops to close to 0 (2.2164e-06).

I wonder if there are ways to improve the model performances, in particular in the area of reducing false positives.

**Comment by svlandeg** [comment]
I guess in an ideal scenario you'd avoid having these garbled texts/sentences in the first place, right? Is there any way you can filter them out? (perhaps as simple as filtering on length?)

If they're not easy to remove, I wonder whether their distribution is higher in the test set than in the training. Perhaps the training data was "cleaned" and the test data wasn't? If so, it may help to add in some garbled sentences + label `NOT_PRESENT` to your training data, to make the classifier robust to those cases.

With respect to being punctuation sensitive, perhaps also there you could think about enriching the training dataset with some "punctuation noise", i.e. adding random punctuation to your examples and keeping the original label intact. Again this should hopefully make your classifier more robust. Ofcourse you'll have to ensure that the overall performance doesn't drop, as punctuation can obviously provide valuable features ;-)
